src/ui/connect/connector.ts

```diff
@@ -1,107 +1,127 @@
+import {isChromium, isFirefox} from 'utils/platform';
 import type {ExtensionData, ExtensionActions, FilterConfig, TabInfo, Message, UserSettings} from '../../definitions';
 
 export default class Connector implements ExtensionActions {
-    private port: chrome.runtime.Port;
-    private counter: number;
+    private changeSubscribers: Set<(data: ExtensionData) => void>;
 
     constructor() {
-        this.counter = 0;
-        this.port = chrome.runtime.connect({name: 'ui'});
+        this.changeSubscribers = new Set();
     }
 
-    private getRequestId() {
-        return ++this.counter;
+    private async sendRequest<T>(request: Message) {
+        return new Promise<T>((resolve, reject) => {
+            chrome.runtime.sendMessage({name: 'ui', ...request}, ({data, error}) => {
+                if (error)
```

Comment:
Braces... Why isn't ESLint erroring on this?

Reply:
Neither `npm run lint` nor `npm run code-style` does anything. I fixed this and similar issues manually.

Reply:
Good to know, more reason to clean up the eslint config.